lib/index.js

```diff
@@ -1723,32 +1740,23 @@ exposeMethod('pluck');
  */
 
 Stream.prototype.pickBy = function (f) {
-
-    return this.consume(function (err, x, push, next) {
+    return this.map(objectOnly(function (x) {
         var out = {};
-        if (err) {
-            push(err);
-            next();
-        }
-        else if (x === nil) {
-            push(err, x);
-        }
-        else if (_.isObject(x)) {
-            for (var k in x) {
-                if (f(k, x[k])) {
-                    out[k] = x[k];
-                }
+        var props = [];  // cache of already seen properties
```

Comment:
Doesn't look like anything is actually pushed to `props`. Also, instead of using an array and `indexOf`(which I assume is a linear operation), should we use a regular object instead (set `props[prop] = true`, check `if (props[prop])`)?

Also, can you add a check for this case?

```javascript
var proto = {a: 1};
var obj = Object.create(proto);
obj.a = 2;

_([obj]).pickBy(function (k, v) {
// check that only called with k === 'a' once.
}).resume();
```